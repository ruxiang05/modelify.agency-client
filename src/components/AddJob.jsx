import React from 'react';
import Page from './Page'

const AddJob = () => (
    <Page>Add Job</Page>
);

export default AddJob;