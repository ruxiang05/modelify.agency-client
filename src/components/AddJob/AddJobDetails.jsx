import React from 'react';
import PropTypes from 'prop-types';
import FormInput from '../FormInput';
import PageHeader from '../PageHeader';

class AddJobDetails extends React.Component {
  constructor(props) {
    super(props);

    this.saveAndContinue = this.saveAndContinue.bind(this);
  }

  saveAndContinue(event) {
    const { nextStep } = this.props;
    event.preventDefault();
    nextStep();
  }

  render() {
    const {
      title, date, address, pay, description, handleChange,
    } = this.props;
    return (
      <div className="page">
        <PageHeader title="Add Job Details" />
        <form>
          <FormInput type="text" value={title} name="title" title="Title" placeholder="Enter job short descritpion" handleChange={handleChange} />
          <FormInput type="date" value={date} name="date" title="Date" placeholder="Enter job date" handleChange={handleChange} />
          <FormInput type="text" value={address} name="address" title="Address" placeholder="Enter job address" handleChange={handleChange} />
          <FormInput type="number" value={pay} name="pay" title="Payment" placeholder="Enter payment amount" handleChange={handleChange} />
          <FormInput type="textarea" value={description} name="description" title="Description" placeholder="Enter job details" handleChange={handleChange} />
          <button type="submit" onClick={this.saveAndContinue}> Save and Continue </button>
        </form>
      </div>
    );
  }
}

AddJobDetails.propTypes = {
  title: PropTypes.string.isRequired,
  date: PropTypes.instanceOf(Date).isRequired,
  address: PropTypes.string.isRequired,
  pay: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  nextStep: PropTypes.func.isRequired,
};

export default AddJobDetails;
