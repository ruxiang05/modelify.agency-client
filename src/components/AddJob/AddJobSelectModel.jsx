import React from 'react';
import ModelList from '../ModelList';
import PageHeader from '../PageHeader';

const AddJobDetails = props => (
  <React.Fragment>
    <PageHeader title="Select Model" />
    <div className="page">
      <ModelList {...props} />
    </div>
  </React.Fragment>
);

export default AddJobDetails;
