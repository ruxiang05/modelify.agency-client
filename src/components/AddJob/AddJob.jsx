import React from 'react';
import PropTypes from 'prop-types';
import AddJobDetails from './AddJobDetails';
import AddJobSelectModel from './AddJobSelectModel';
import { SelectModelProvider } from '../../contexts/SelectModelContext';
import api from '../../api';
import { getToken } from '../../auth';

class AddJob extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      step: 1,
      model: '',
      title: '',
      date: '',
      address: '',
      pay: '',
      description: '',
    };
    this.nextStep = this.nextStep.bind(this);
    this.prevStep = this.prevStep.bind(this);
    this.selectModel = this.selectModel.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  nextStep() {
    this.setState(prevState => ({
      step: prevState.step + 1,
    }));
  }

  prevStep() {
    this.setState(prevState => ({
      step: prevState.step - 1,
    }));
  }

  selectModel(value) {
    const { history } = this.props;
    this.setState({
      model: value,
    }, () => {
      api.jobs.addJob(getToken(), this.state).then(() => history.push('/jobs'));
    });
  }

  handleChange(event) {
    const { name, value } = event.target;
    this.setState(prevState => ({ ...prevState, [name]: value }));
  }

  /* eslint-disable consistent-return,default-case */
  render() {
    const { step, ...values } = this.state;

    switch (step) {
      case 1: return <AddJobDetails {...values} handleChange={this.handleChange} nextStep={this.nextStep} />;
      case 2: return <SelectModelProvider value={{ selectModel: this.selectModel }}><AddJobSelectModel {...values} prevStep={this.prevStep} /></SelectModelProvider>;
    }
  }
}

AddJob.propTypes = {
  history: PropTypes.shape({}).isRequired,
};

export default AddJob;
