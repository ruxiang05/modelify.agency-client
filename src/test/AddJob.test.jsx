/* Uses methods and/or components from react and enzyme */
import React from 'react';
import { shallow } from 'enzyme';
import { createBrowserHistory } from 'history';
import AddJob from '../components/AddJob/AddJob';

jest.mock('../api.js');

const history = createBrowserHistory({
  basename: '/',
  keyLength: 0,
});

let wrapper;

beforeEach(() => {
  wrapper = shallow(<AddJob history={history} />);
});

describe('AddJob', () => {
  it('should render AddJob', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should go to next step', () => {
    expect(wrapper.state().step).toBe(1);
    wrapper.instance().nextStep();
    expect(wrapper.state().step).toBe(2);
  });

  it('should go to previous step', () => {
    wrapper.setState({ step: 2 });
    wrapper.instance().prevStep();
    expect(wrapper.state().step).toBe(1);
  });

  it('should select model and add job', async () => {
    history.push('/add-job');
    await wrapper.instance().selectModel('model');
    expect(wrapper.state().model).toBe('model');
    expect(history.location.pathname).toBe('/jobs');
  });

  it('should handleChange correctly', async () => {
    const event = {
      target: {
        name: 'title',
        value: 'some title',
      },
    };
    await wrapper.instance().handleChange(event);
    expect(wrapper.state().title).toBe('some title');
  });
});
