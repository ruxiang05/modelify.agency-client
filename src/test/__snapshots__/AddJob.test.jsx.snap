// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddJob should render AddJob 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddJob
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "address": "",
      "date": "",
      "description": "",
      "handleChange": [Function],
      "model": "",
      "nextStep": [Function],
      "pay": "",
      "title": "",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "address": "",
        "date": "",
        "description": "",
        "handleChange": [Function],
        "model": "",
        "nextStep": [Function],
        "pay": "",
        "title": "",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
